### DISCUSSION TITLE
Using wasm with my extension

### DISCUSSION THREAD

**Discussion by tmirza** [eclipse-theia/theia#11362] [Q&A] [answered]
Hi Guys, 

I have a quick question about using wasm with Theia. I have my native C++ application that I want to use with my theia IDE running in browser. I used emscripten to build WASM file of my application. I can run my wasm from java script file. For Theia, I added code to load wasm file and call exported function from sample menu command handler. When I run my browser app, it gives error GET http://localhost:3000/myCustom.wasm 404 (Not Found) 
I want to know where should I place my wasm file and how should I specify its path for loading. Also, is there any example of using WASM with theia that I can use as reference.

Thanks

**Comment by msujew** [comment, marked as answer]
Hi @tmirza,

there are basically 2 ways to resolve this:

1. The simple one: You just need to place the `myCustom.wasm` file in the `lib` directory of your browser app (where also the `index.html` and `bundle.js` are).
2. The more complex one: You can create a new express middleware using the `BackendApplicationContribution#configure` method and load the file this way. This method does not rely on a specific path of the `myCustom.wasm` file.

**Reply by tmirza** [reply]
Thanks @msujew for quick reply.
